internal/engine/planner.go

```diff
@@ -569,17 +569,19 @@ func evalComprehensionBody(env *cel.Env, pvars interpreter.PartialActivation, e
 		}
 		updateIds(le)
 		ast := cel.ParsedExprToAst(&exprpb.ParsedExpr{Expr: le})
-		partialVars, err := cel.PartialVars(pvars, cel.AttributePattern(ce.IterVar))
+
+		unknowns := append(pvars.UnknownAttributePatterns(), cel.AttributePattern(ce.IterVar))
```

Comment:
If I'm reading above correctly, we don't allow just any unknown variable, it should just be `request.resource` or `R`, right? I think the same should apply here - if you introduce a random `x` that should be an error, but accessing a resource attribute should return an AST.

Reply:
Short version: Lambda parameter `x` is unknown in the context of lambda body evaluation. For example, `P.roles.exists(x, x.startsWith("admin"))`. Inside the inner expression `x.startsWith("admin")` the `x` is unknown.

Longer version:
When `evalComprehensionBody` is invoked the first time, the `pvars` contains `R`, `request`, `request.resource` in the unknowns list. In this code block, `ce` is a comprehension expression. 
For example `ce = P.roles.exists(x, x.startsWith("admin"))`, `le = x.startsWith("admin")`, `ce.IterVar = x`. In other words. `ce` is lambda, `x` is a lambda parameter, `le` is a lambda body.
We evaluate the lambda body here, so we must add `x` to the list of unknowns.

Reply:
Thanks, I was confused about the purpose of `pvars.UnknownAttributePatterns()` but it makes sense now 🙂